test/TokenRequestTest.js

```diff
@@ -1,51 +1,340 @@
+
+        const actualUserMiniMeBalance = await tokenManager.spendableBalanceOf(rootAccount)
+        const actualVaultBalance = await vault.balance(mockErc20.address)
+
+        assert.equal(actualUserMiniMeBalance, expectedUserMiniMeBalance)
+        assert.equal(actualVaultBalance, expectedVaultBalance)
+      })
+
+      it('finalise token request (ETH)', async () => {
+        const expectedUserMiniMeBalance = 300
+        const expectedVaultBalance = 200
+
+        await tokenRequest.createTokenRequest(ETH_ADDRESS, expectedVaultBalance, expectedUserMiniMeBalance, {
+          from: rootAccount,
+          value: expectedVaultBalance,
+        })
+
+        await forwarderMock.forward(script, { from: rootAccount })
+
+        const actualUserMiniMeBalance = await tokenManager.spendableBalanceOf(rootAccount)
+        const actualVaultBalance = await vault.balance(ETH_ADDRESS)
+
+        assert.equal(actualUserMiniMeBalance, expectedUserMiniMeBalance)
+        assert.equal(actualVaultBalance, expectedVaultBalance)
+      })
+      it('it should not finalise the same request twice', async () => {
+        const expectedUserMiniMeBalance = 300
+        const expectedVaultBalance = 200
+
+        await tokenRequest.createTokenRequest(ETH_ADDRESS, expectedVaultBalance, expectedUserMiniMeBalance, {
+          from: rootAccount,
+          value: expectedVaultBalance,
+        })
+
+        await tokenRequest.createTokenRequest(ETH_ADDRESS, expectedVaultBalance, expectedUserMiniMeBalance, {
+          from: rootAccount,
+          value: expectedVaultBalance,
+        })
+
+        await forwarderMock.forward(script, { from: rootAccount })
+
+        await assertRevert(forwarderMock.forward(script, { from: rootAccount }), 'TOKEN_REQUEST_NO_DEPOSIT')
+      })
+    })
+
+    describe('refundTokenRequest(uint256 _tokenRequestId) ', () => {
+      const refundEthAccount = accounts[2]
+      it('refund token (ERC20)', async () => {
+        const refundAmount = 100
+        const expectedUserBalance = await mockErc20.balanceOf(rootAccount)
+
+        await mockErc20.approve(tokenRequest.address, refundAmount, {
+          from: rootAccount,
+        })
+        await tokenRequest.createTokenRequest(mockErc20.address, refundAmount, 1, {
+          from: rootAccount,
+        })
+
+        await tokenRequest.refundTokenRequest(0, { from: rootAccount })
+
+        const actualUserBalance = await mockErc20.balanceOf(rootAccount)
+        assert.equal(Number(actualUserBalance), Number(expectedUserBalance))
+      })
+
+      it('refund ETH', async () => {
+        const weiValue = 3000000000000000
+        const expectedETHBalance = await web3.eth.getBalance(refundEthAccount)
+
+        const request = await tokenRequest.createTokenRequest(ETH_ADDRESS, weiValue, 1, {
+          value: weiValue,
+          from: refundEthAccount,
+        })
+
+        const requestTransaction = await web3.eth.getTransaction(request.tx)
+        let requestPrice = new BN()
```

Comment:
You can remove this line and declare `requestPrice` as a const where it is used.